meson.build

```
@@ -0,0 +1,355 @@
+]
+
+conf.set('SIZEOF_UNSIGNED_LONG', cc.sizeof('unsigned long'))
+
+if host_system != 'windows'
+  pthread_create_deps = []
+  if not cc.has_function('pthread_create')
+    libpthread = cc.find_library('pthread', required: false)
+    if cc.has_function('pthread_create', dependencies: libpthread)
+      pthread_create_deps += libpthread
+    else
+      error('could not find pthread_create')
+    endif
+  endif
+
+  if not cc.has_function('nanosleep')
+    nanosleep_deps = cc.find_library('rt', required: false)
+    if not cc.has_function('nanosleep', dependencies: nanosleep_deps)
+      error('could not find nanosleep')
+    endif
+  endif
+
+  libdl_deps = []
+  if not cc.has_function('dlopen')
+    libdl = cc.find_library('dl', required: false)
+    if cc.has_function('dlopen', dependencies: libdl)
+      libdl_deps += libdl
+    else
+      error('could not find dlopen')
+    endif
+  endif
+
+  # for Solaris we need -lsocket -lnsl for socket stuff, gethostbyname
+  # is just a dummy to find -lnsl
+  libnsl = cc.find_library('nsl', required: false)
+  if libnsl.found()
+    cc.has_function('gethostbyname', dependencies: libnsl)
+  endif
+
+  libsocket = cc.find_library('socket', required: false)
+  if libsocket.found()
+    if not cc.has_function('connect', dependencies: libsocket)
+      error('could not find socket')
+    endif
+  endif
+
+  if cc.has_header('locale.h')
+    conf.set('HAVE_LOCALE_H', 1)
+    if cc.has_header_symbol('locale.h', 'locale_t')
+      conf.set('HAVE_LOCALE_T', 1)
+      if cc.has_function('newlocale', prefix: '#include <locale.h>')
+        conf.set('HAVE_NEWLOCALE', 1)
+      endif
+      if cc.has_function('strerror_l', prefix: '#include <string.h>')
+        conf.set('HAVE_STRERROR_L', 1)
+      endif
+    endif
+  endif
+
+  # These are things we can work around
+  headers = [
+    'sys/resource.h',
+    'ucred.h'
+  ]
+
+  foreach h : headers
+    if cc.has_header(h)
+      conf.set('HAVE_' + h.underscorify().to_upper(), 1)
+    endif
+  endforeach
+
+  functions = [
+    'asprintf',
+    'basename',
+    'fdwalk',
+    'getauxval',
+    'getexecname',
+    'getpeereid',
+    'getpeerucred',
+    'getprogname',
+    'getresuid',
+    'issetugid',
+    'memdup',
+    'mkdtemp',
+    'mkstemp',
+    'reallocarray',
+    'secure_getenv',
+    'setenv',
+    'strerror_r',
+    'strndup',
+    'strnstr',
+    'vasprintf',
+  ]
+
+  foreach f : functions
+    if cc.has_function(f)
+      conf.set('HAVE_' + f.underscorify().to_upper(), 1)
+    endif
+  endforeach
+
+  if cc.has_header_symbol('stdio.h', 'asprintf', prefix: '#define _GNU_SOURCE')
+    conf.set('HAVE_DECL_ASPRINTF', 1)
+  else
+    conf.set('HAVE_DECL_ASPRINTF', 0)
+  endif
+
+  if cc.has_header_symbol('stdio.h', 'vasprintf', prefix: '#define _GNU_SOURCE')
+    conf.set('HAVE_DECL_VASPRINTF', 1)
+  else
+    conf.set('HAVE_DECL_VASPRINTF', 0)
+  endif
+
+  if cc.has_header_symbol('stdlib.h', 'reallocarray')
+    conf.set('HAVE_DECL_REALLOCARRAY', 1)
```

You can also use `set10` here:

```suggestion
    conf.set10('HAVE_DECL_REALLOCARRAY', cc.has_header_symbol('stdlib.h', 'reallocarray'))
```